.github/workflows/build_publish_lambda_layer.yml

```diff
@@ -65,6 +65,72 @@ jobs:
           name: ruby-layer-${{ matrix.arch }}.zip
           path: lambda/build/ruby-layer-${{ matrix.arch }}.zip
 
+  # extract the built layer from artifacts, then scan it with reverselab
+  reverselab_scan_layer:
+    needs:
+      - build_layer
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - x86_64
+          - arm64
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - if: ${{ inputs.publish-dest == 'staging' }}
+        name: use staging credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.LAMBDA_PUBLISHER_ARN_STAGING }}
+          aws-region: us-east-1
+
+      - if: ${{ inputs.publish-dest == 'production' }}
+        name: use production credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.LAMBDA_PUBLISHER_ARN_PROD }}
+          aws-region: us-east-1
+
+      - name: extract layer zip from artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ruby-layer-${{ matrix.arch }}.zip
+          path: lambda
+
+      - name: extract current solarwinds_apm version
+        run: |
+          APM_VERSION=$(grep "gem 'solarwinds_apm'" lambda/otel/layer/Gemfile | awk -F"'" '{print $4}')
+          APM_VERSION="${APM_VERSION//./_}"
+          LAYER_NAME=solarwinds-apm-ruby-${{ matrix.arch }}-$APM_VERSION
+          LATEST_VERSION=$(aws lambda list-layer-versions --layer-name $LAYER_NAME | jq -r '.LayerVersions | max_by(.Version) | .Version')
+          NEXT_LATEST_VERSION=$(( $LATEST_VERSION + 1))
+          echo "LAYER_VERSION=$NEXT_LATEST_VERSION" >> $GITHUB_ENV
```

Comment:
Oh i see this is where you're using the AWS credentials... not sure whether the layer version will really be meaningful (since it could be different per region). Can we set the version as solarwinds_apm version instead? Take a look at the RL scans for Python for an idea.

I know this assumes that Lambda layers are published in lockstep with main library release, but that likely will be the case (and is what other languages are doing).

Reply:
Updated.